test/extended/networking/services.go

```diff
@@ -0,0 +1,67 @@
+package networking
+
+import (
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/test/e2e"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("[networking] services", func() {
+	f1 := e2e.NewFramework("net-services2")
+	f2 := e2e.NewFramework("net-services2")
+
+	It("should allow connections to another pod on the same node via a service IP", func() {
+		Expect(checkService(f1, f1, 1)).To(Succeed())
+	})
+
+	It("should allow connections to another pod on a different node via a service IP", func() {
+		Expect(checkService(f1, f1, 2)).To(Succeed())
+	})
+
+	Specify("single-tenant plugins should allow connections to pods in different namespaces on the same node via service IPs", func() {
+		skipIfMultiTenant()
+		Expect(checkService(f1, f2, 1)).To(Succeed())
+	})
+
+	Specify("single-tenant plugins should allow connections to pods in different namespaces on different nodes via service IPs", func() {
+		skipIfMultiTenant()
+		Expect(checkService(f1, f2, 2)).To(Succeed())
+	})
+
+	Specify("multi-tenant plugins should prevent connections to pods in different namespaces on the same node via service IPs", func() {
+		skipIfSingleTenant()
+		err := checkService(f1, f2, 1)
+		Expect(err).To(HaveOccurred())
+	})
+
+	Specify("multi-tenant plugins should prevent connections to pods in different namespaces on different nodes via service IPs", func() {
+		skipIfSingleTenant()
+		err := checkService(f1, f2, 2)
+		Expect(err).To(HaveOccurred())
+	})
+})
+
+func checkService(serverFramework, clientFramework *e2e.Framework, numNodes int) error {
```

Comment:
(No action required) Consider accepting a boolean (e.g. multiNode) instead of an int to indicate whether the check will be performed against multiple nodes.